sound/soc/sof/ipc4-topology.c

```diff
@@ -1807,7 +1807,7 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 	struct sof_ipc4_copier_data *copier_data;
 	int input_fmt_index, output_fmt_index;
-	struct snd_pcm_hw_params ref_params;
+	struct snd_pcm_hw_params ref_params = { 0 };
```

Comment:
This is so false report, I have refused to fix it out of principle ;)
ref_params _is_ going to be initialized and when it is not we will just return with -EINVAL.
See the first switch (swidget->id) cases...

Reply:
Yes, this is a false positive. I wonder if we can mark it somehow so that the CI report

```
 build test / GCC static -fanalyzer (pull_request) Failing after 3m 
 ```
 
 would be useful.

Reply:
if nothing else we should take this patch..

Reply:
@ujfalusi i noticed that there are at least 3-4 cases of  similar false positives mostly in @thesofproject/amd  code. 

I have no strong preference or merging this patch.

Cc: @ranj063 @lgirdwood  if you consider this useful you can merge it.

Reply:
@dbaluta, fwiw, with
```
$ gcc --version                                 
gcc (GCC) 14.2.1 20240910
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
-fanalyzer no longer reports false positive for this.

Reply:
@ujfalusi that's good news. 

We are already using latest ubuntu, so at some point in the future the compiler version will be upgraded

Would anyone care to use a newer compiler version in  .github/workflows/buildtest.yml?

Reply:
I wish we can switch to newer compiler version. :)

Reply:
I'll try to figure this out